lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -640,36 +640,40 @@ static void AddRequiredAliases(Block *block, lldb::StackFrameSP &stack_frame_sp,
     swift::BoundGenericEnumType *optional_type =
         referent_type->getAs<swift::BoundGenericEnumType>();
 
-    if (!optional_type)
-      return;
+    if (!optional_type || optional_type->getGenericArgs().empty())
+      return false;
 
     swift::Type first_arg_type = optional_type->getGenericArgs()[0];
 
-    swift::ClassType *self_class_type =
-        first_arg_type->getAs<swift::ClassType>();
-
-    if (!self_class_type)
-      return;
+    // Only class types can be weakly captured.
+    if (!llvm::isa<swift::ClassType>(first_arg_type) &&
+        !llvm::isa<swift::BoundGenericClassType>(first_arg_type))
```

Comment:
If the first condition is true, won't the second always be false?

Reply:
There's a `!` there

The condition is "if the type is not a class nor a bound generic class, return early".